Finder.php

```diff
@@ -134,7 +134,19 @@ public function in($path)
         }
 
         foreach ($path as $p) {
-            $this->_paths[] = $p;
+            if( 1 === preg_match('/\*|\?|\[([^\]]+)\]|\{([^}]+)\}/', $p) ) {
```

Comment:
`{` and `}` can be removed from the pattern then.
Also, no need to check if `[` has a closing `]`.
Finally, you can simplify by using a class instead of a disjunction: `/[\*\?\[\]/`.